public/js/app.Ctrl.js

```diff
@@ -0,0 +1,201 @@
+App.controller('App.Ctrl', function($scope, $timeout, AppService) {
+    const colorCode = {1:"green", 2: "red", 3: "yellow", 4: "blue"};
+    const speed = 500; // in milliseconds
+    var on = false;
+    var lock = false;
+    var strict = false;
+    var colorSequence = [];
+    var count = 0;
+    var round = 1;
+    var turnOffTimer;
+    var recursiveTimer;
+    var timer;
+    var sound = {1:{}, 2:{}, 3: {}, 4: {}};
+    var buzzer;
+    var win;
+
+    $scope.init = function() {
+        if(on === true) {
+            setUpSoundEffects();
+            reset(true);
+        }
+    };
+
+    $scope.toggleOnOff = function() {
+        if(on === true) {
+            on = false;
+            $scope.btn__off = {"background-color": "#8A96B6"};
+            $scope.btn__on = {"background-color": "#333"};
+            cancelAllTimers();
+            turnOffAllLights();
+            $scope.screen = "";
+        }
+        else {
+            on = true;
+            $scope.btn__off = {"background-color": "#333"};
+            $scope.btn__on = {"background-color": "#8A96B6"};
+        }
+    };
+
+    $scope.toggleStrict = function() {
+        if(strict === false) {
+            strict = true;
+            $scope.btn__strict = {"box-shadow":"inset 0 0 8px #666"};
+        }
+        else {
+            strict = false;
+            $scope.btn__strict = {"box-shadow":"inset 0 0 1px #666"}
+        }
+
+    };
+
+    startRound = function(round) {
+        $scope.screen = round;
+        var color = colorSequence[count];
+        playSound(color.code);
+        // Light up the color by changing its CSS background-color value
+        $scope[colorCode[color.code]] = color.lightColorObj;
+        turnOffTimer = $timeout(function() {
+            // darken the color by changing back its CSS background-color value
+            $scope[colorCode[color.code]] = color.darkColorObj;
+            recursiveTimer = $timeout(function(){
+                count++;
+                if(count < round)
+                    startRound(round);
+            }, speed/2);
+        }, speed);
+    };
+
+     $scope.passUserInput = function(n) {
+         if(on === true && lock === false) {
+             cancelAllTimers();
+             turnOffAllLights();
+             var result = AppService.addUserInput(n, round);
+             switch(result) {
+                 // user input is incorrect
```

Comment:
Using integer values to reflect state make it hard for people to understand your program.  What was 1 again?  Declare and use constants for stuff like this.  STATE.WIN = 'win'.  case STATE.WIN: